libfwupdplugin/fu-bios-settings.c

```diff
@@ -615,6 +617,60 @@ fu_bios_settings_from_json(FuBiosSettings *self, JsonNode *json_node, GError **e
 	return TRUE;
 }
 
+/**
+ * fu_bios_settings_from_json_file:
+ * @self: A #FuBiosSettings
+ * @fn: a path to a JSON file
+ * @error: (nullable): optional return location for an error
+ *
+ * Adds all BIOS attributes from a JSON filename
+ *
+ * Returns: TRUE for success, FALSE for failure
+ *
+ * Since: 1.8.4
+ **/
+gboolean
+fu_bios_settings_from_json_file(FuBiosSettings *self, const gchar *fn, GError **error)
+{
+	g_autofree gchar *data = NULL;
+	g_autoptr(JsonParser) parser = json_parser_new();
+
+	if (!g_file_get_contents(fn, &data, NULL, error))
+		return FALSE;
+	if (!json_parser_load_from_data(parser, data, -1, error)) {
+		g_prefix_error(error, "%s doesn't look like JSON data: ", fn);
+		return FALSE;
+	}
+	return fu_bios_settings_from_json_node(self, json_parser_get_root(parser), error);
+}
+
+/**
+ * fu_bios_settings_to_hash_table:
+ * @self: A #FuBiosSettings
+ *
+ * Creates a #GHashTable with the name and current value of
+ * all BIOS settings.
+ *
+ * Returns: (transfer full): name/value pairs
+ * Since: 1.8.4
+ **/
+GHashTable *
+fu_bios_settings_to_hash_table(FuBiosSettings *self)
```

Comment:
I think `fu_bios_settings_to_hash_kv` might be better.

Reply:
ack